### PR TITLE
Tolerate white-spaces on finish line beginning

### PR DESCRIPTION
**Pull request by pdecat**
This relaxes finish line matching to support for white-spaces at the beginning and should partly address #11 (it still requires the finish line to not contain terraform code).

Maybe this could go a bit further and remove the spaces at the beginning of the matched finish line.

### CODE REVIEW

**Comment by katbyte** [comment]
Hey @pdecat,

Thanks for this PR. I want to accept it but we (azurerm) actually rely on that behavior to enforce a consistent formatting for our acceptance tests so doing a pattern match is not ideal. However what we could do is expose a flag allowing for custom start/end strings to be passed in, maybe something like ```terrafmt -blockstart '    `,'```?

WDYT

**Comment by pdecat** [comment]
Hi @katbyte, wouldn't removing the spaces at the beginning of the matched finish line address this?

**Comment by pdecat** [comment]
Note: I too want to enforce a consistent formatting for our zabbix provider acceptance tests aligned on the official providers.

It's just that currently, terrafmt skips most of them because of those spaces.

**Comment by katbyte** [comment]
@pdecat, yep, if you remove the spaces at the of the start of that last line it should work. in azurerm everything is:
```
	return fmt.Sprintf(`
%s

data "azurerm_kubernetes_cluster" "test" {
  name                = "${azurerm_kubernetes_cluster.test.name}"
  resource_group_name = "${azurerm_kubernetes_cluster.test.resource_group_name}"
}
`, r)
```

One quirk of terrafmt is it will not format the start or finish lines and doesn't consider them part of the tf block. Mainly because it would be much harder and we start having to think of a byte stream rather then a list of lines. For azurerm we did mass find/replaces in GoLand to fix a lot of issue where the closing quote is not on its own line.

**Comment by pdecat** [comment]
I've updated the PR to remove the leading spaces on the finish lines:

```sh
# git reset --hard
HEAD is now at 2962e7f Merge pull request #7 from claranet/claranet
# terrafmt fmt zabbix/resource_zabbix_item_test.go
# git diff zabbix/resource_zabbix_item_test.go
diff --git zabbix/resource_zabbix_item_test.go zabbix/resource_zabbix_item_test.go
index 54a2016..51d533b 100644
--- zabbix/resource_zabbix_item_test.go
+++ zabbix/resource_zabbix_item_test.go
@@ -51,52 +51,52 @@ func TestAccZabbixItem_Basic(t *testing.T) {
 
 func testAccZabbixItemConfig(groupName string, templateName string, itemName string) string {
 	return fmt.Sprintf(`
-		resource "zabbix_host_group" "zabbix" {
-			name = "%s"
-		}
+resource "zabbix_host_group" "zabbix" {
+  name = "%s"
+}
 
-		resource "zabbix_template" "my_zbx_template" {
-			host = "%s"
-			groups = ["${zabbix_host_group.zabbix.name}"]
-			name = "display name %s"
-			description = "description for template %s"
-	  	}
-	  
-		resource "zabbix_item" "my_item1" {
-			name = "%s"
-			key = "bilou.bilou"
-			delay = "34"
-			description = "description for item : %s"
-			trends = "300"
-			history = "25"
-			host_id = "${zabbix_template.my_zbx_template.id}"
-	  	}
-	`, groupName, templateName, templateName, templateName, itemName, itemName)
+resource "zabbix_template" "my_zbx_template" {
+  host        = "%s"
+  groups      = ["${zabbix_host_group.zabbix.name}"]
+  name        = "display name %s"
+  description = "description for template %s"
+}
+
+resource "zabbix_item" "my_item1" {
+  name        = "%s"
+  key         = "bilou.bilou"
+  delay       = "34"
+  description = "description for item : %s"
+  trends      = "300"
+  history     = "25"
+  host_id     = "${zabbix_template.my_zbx_template.id}"
+}
+`, groupName, templateName, templateName, templateName, itemName, itemName)
 }
 
 func testAccZabbixItemUpdate(groupName string, templateName string, itemName string) string {
 	return fmt.Sprintf(`
-		resource "zabbix_host_group" "zabbix" {
-			name = "%s"
-		}
+resource "zabbix_host_group" "zabbix" {
+  name = "%s"
+}
 
-		resource "zabbix_template" "my_zbx_template" {
-			host = "%s"
-			groups = ["${zabbix_host_group.zabbix.name}"]
-			name = "display name %s"
-			description = "description for template %s"
-	  	}
-	  
-		resource "zabbix_item" "my_item1" {
-			name = "update_%s"
-			key = "update.bilou.bilou"
-			delay = "23"
-			description = "update description for item : %s"
-			trends = "3"
-			history = "2"
-			host_id = "${zabbix_template.my_zbx_template.id}"
-	  	}
-	`, groupName, templateName, templateName, templateName, itemName, itemName)
+resource "zabbix_template" "my_zbx_template" {
+  host        = "%s"
+  groups      = ["${zabbix_host_group.zabbix.name}"]
+  name        = "display name %s"
+  description = "description for template %s"
+}
+
+resource "zabbix_item" "my_item1" {
+  name        = "update_%s"
+  key         = "update.bilou.bilou"
+  delay       = "23"
+  description = "update description for item : %s"
+  trends      = "3"
+  history     = "2"
+  host_id     = "${zabbix_template.my_zbx_template.id}"
+}
+`, groupName, templateName, templateName, templateName, itemName, itemName)
 }
 
 func testAccZabbixItemExists(resource string) resource.TestCheckFunc {
```

File: lib/blocks/blockreader.go [hunk 1]
```diff
@@ -107,7 +109,7 @@ func (br *Reader) DoTheThing(filename string) error {
 		l := s.Text() + "\n"
 
 		if err := br.LineRead(br, br.LineCount, l); err != nil {
-			return fmt.Errorf("NB LineRead failed @ %s:%d for %s: %v", br.FileName, br.LineCount, l, err)
+			return fmt.Errorf("NB LineRead failed @ %s#%d for %s: %v", br.FileName, br.LineCount, l, err)
```
Comment (katbyte) [inline lib/blocks/blockreader.go]: This is reverting a PR to change to :s as such could we revert this change?

File: lib/blocks/blockreader.go [hunk 2]
```diff
@@ -57,9 +58,8 @@ func IsStartLine(line string) bool {
 }
 
 func IsFinishLine(line string) bool {
-	if line == "`)\n" { // acctest
-		return true
-	} else if strings.HasPrefix(line, "`,") { // acctest
+	spaceLeftAccTestMatcher := regexp.MustCompile("^[[:space:]]*`(,|\\)\n)")
```
Comment (katbyte) [inline lib/blocks/blockreader.go]: Could we make this a constant somewhere?

File: lib/blocks/blockreader.go (hunk 1 above)
Comment (pdecat) [reply lib/blocks/blockreader.go]: Oops, indeed! Fixed.

What about https://github.com/katbyte/terrafmt/pull/12/files#diff-9e4bd7d4c75f212c91e8d3bc2283ba3aR134 ?

File: lib/blocks/blockreader.go (hunk 2 above)
Comment (pdecat) [reply lib/blocks/blockreader.go]: Done.

**Comment by pdecat** [comment]
Great idea! I've added a command line flag to opt in fixing the finish lines! Named it `--fix-finish-lines`, plural, as a single fmt pass will likely format more than one block.
